plutus-ledger-api/src/PlutusLedgerApi/V1/Credential.hs

```diff
@@ -17,15 +17,20 @@ import Control.DeepSeq (NFData)
 import GHC.Generics (Generic)
 import PlutusLedgerApi.V1.Crypto (PubKeyHash)
 import PlutusLedgerApi.V1.Scripts (ValidatorHash)
-import PlutusTx qualified as PlutusTx
+import PlutusTx qualified
 import PlutusTx.Bool qualified as PlutusTx
 import PlutusTx.Eq qualified as PlutusTx
 import Prettyprinter (Pretty (..), (<+>))
 
--- | Staking credential used to assign rewards
+-- | Staking credential used to assign rewards.
 data StakingCredential
+    -- | The staking hash is the `Credential` required to unlock a transaction output. Either
+    -- a public key credential (`Crypto.PubKeyHash`) or
+    -- a script credential (`Scripts.ValidatorHash`). Both are hashed with /BLAKE2b-244/. 28 byte.
     = StakingHash Credential
-    | StakingPtr Integer Integer Integer -- NB: The fields should really be Word64 / Natural / Natural, but 'Integer' is our only integral type so we need to use it instead.
+    -- | NB: The fields should really be `Word64` `Natural` `Natural`,
```

Comment:
Do you want to say whwat the word64 , natural , natural represent? I think it is the slot number, the transaction index inside that slot and the txout index inside that transaction. Is that correct @JaredCorduan  ?

Reply:
Actually after reading the shelley spec, the `Word64` may even be wrong, because under "B.4 Payloads" of https://hydra.iohk.io/build/16861845/download/1/ledger-spec.pdf    it says :

> pointer address one 28-byte bytestring, and **three variable-length** unsigned integers

Reply:
Yes I would like to put in more info if we have it.

Reply:
The triple represents: the slot number, the transaction index (within the block), and the certificate index (within the transaction).

In the spec, "three variable-length unsigned integers" refers to the wire spec, not the haskell types.

The implementation uses a `Word64` for all three of these.